### DISCUSSION TITLE
How to insert rather than update

### DISCUSSION THREAD

**Discussion by adminSxs** [sqlalchemy/sqlalchemy#8460] [Usage Questions]
### Describe the use case

I have two datasource, db1,db2

```python
db_urls = {
    "db1": "mysql+pymysql://root:12345@127.0.0.1:3306/demo1?charset=utf8mb4",
    "db2":"mysql+pymysql://root:12345@127.0.0.1:3306/demo2?charset=utf8mb4"
};

def get_session(env):
    engine = create_engine(db_urls[env] )
    Session = sessionmaker(bind=engine)
    return scoped_session(Session)

class Test(Base):
    __tablename__ = "test"
    id = Column(BigInteger, primary_key=True)
   name=Column(String)
  
   def clone(self):
        return deepcopy(self) 
 

```

find data in db1,and insert to db2
```python
 db1_session = get_session(db1);
 db2_session = get_session(db2);

  result = db1_session.query(Test).all();
    
  for data in result:
      new_data = data.clone()
     db2_session.add(new_data)
 db2_session.commit()
 db1_session.close()
 db2_session.close()
```

exeute aborve code, error `sqlalchemy.orm.exc.StaleDataError: UPDATE statement on table 'test' expected to update 2 row(s); 0 were matched.`

why is update?

### Databases / Backends / Drivers targeted

Mysql

### Example Use

i want to insert success

**Comment by zzzeek** [comment]
hey there -

it's an UPDATE because when you load an object it's in the [persistent state](https://docs.sqlalchemy.org/en/14/glossary.html#term-persistent) and has an [identity key](https://docs.sqlalchemy.org/en/14/orm/internals.html?highlight=persistent#sqlalchemy.orm.InstanceState.identity) indicating that it is an existing row in its table.

use [make_transient](https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=make_transient#sqlalchemy.orm.make_transient) 

```py
from sqlalchemy.orm import make_transient

new_data = data.clone()
make_transient(new_data)
new_session.add(new_data)
```
